src/lib.rs

```diff
@@ -53,6 +55,8 @@ pub fn check_links(ctx: &RenderContext) -> Result<(), Error> {
         }
     }
 
+    rayon::ThreadPoolBuilder::new().num_threads(6).build_global().unwrap();
```

Comment:
You don't usually need to manually instantiate the rayon thread pool, do you? Also, just out of curiosity what makes you choose 6 threads?

Reply:
You need to instantiate it if you want to set the concurrency level (currently it defaults to ncpus).  I arbitrarily picked 6 since that is the default per-server concurrency that Firefox uses.  I think trying to track per-server concurrency would be a much more substantial change, but would make it even faster.